ui/StatusQ/src/StatusQ/Controls/StatusImageSelector.qml

```diff
@@ -0,0 +1,162 @@
+
+        spacing: 16
+
+        StatusBaseText {
+            id: label
+
+            elide: Text.ElideRight
+            font.pixelSize: Theme.primaryTextFontSize
+        }
+
+        Rectangle {
+            id: imageSelector
+
+            Layout.fillWidth: true
+            Layout.fillHeight: true
+            radius: 8
+            color: Theme.palette.baseColor2
+
+            StatusRoundButton {
+                id: addButton
+
+                z: parent.z + 1
+                icon.name: "add"
+                x: imageSelector.width - 2 * addButton.width / 3 - d.buttonsInsideOffset
+                y: - addButton.height / 3 + d.buttonsInsideOffset
+                type: StatusRoundButton.Type.Secondary
+                onClicked: fileDialog.open()
+            }
+
+            ColumnLayout {
+                id: uploadTextPanel
+
+                width: parent.width
+                anchors.centerIn: parent
+
+                StatusIcon {
+                    icon: "images_icon"
+                    Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+                    Layout.preferredWidth: 20
+                    Layout.preferredHeight: 18
+                    color: Theme.palette.baseColor1
+                    fillMode: Image.PreserveAspectFit
+                }
+
+                StatusBaseText {
+                    id: uploadText
+
+                    Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+                    Layout.topMargin: 5
+                    Layout.preferredWidth: parent.width - 2 * d.imageSelectorPadding
+                    font.pixelSize: Theme.primaryTextFontSize
+                    color: Theme.palette.baseColor1
+                    wrapMode: Text.WordWrap
+                    lineHeight: 1.2
+                    horizontalAlignment: Text.AlignHCenter
+                }
+
+                StatusBaseText {
+                    id: additionalText
+
+                    visible: !!root.additionalText
+                    Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+                    Layout.preferredWidth: parent.width - 2 * d.imageSelectorPadding
+                    font.pixelSize: Theme.secondaryTextFontSize
+                    color: Theme.palette.baseColor1
+                    wrapMode: Text.WordWrap
+                    lineHeight: 1.2
+                    horizontalAlignment: Text.AlignHCenter
+                }
+            }
+
+            Image {
+                id: image
+
+                anchors.fill: parent
+                fillMode: Image.PreserveAspectFit
+            }
+        }
+    }
+
+    FileDialog {
+        id: fileDialog
+
+        folder: shortcuts.pictures
+        nameFilters: [ qsTr("Supported image formats (%1)").arg("*.jpg *.jpeg *.jfif *.webp *.png *.heif") ]
+        onAccepted: {
+            if (fileDialog.fileUrls.length > 0) {
+                image.source = fileDialog.fileUrls[0]
+                uploadTextPanel.visible = false
```

Comment:
`image.source` and `uploadTextPanel.visible` better be just bindings

Reply:
Not sure how to improve it by using just bindings. Could you suggest me something?!

Reply:
Yeah, nevermind, just keep it; it wouldn't be much better :)